camelot/view/action_steps/item_view.py

```diff
@@ -294,36 +299,31 @@ def get_progress_dialog(self):
                 continue
             new_model.add_action_route(tuple(action['route']))
 
-        response = qml_action_step(list_gui_context, 'OpenTableView',
-                serialized_step, { 'model': new_model })
-        context_id = response['context_id']
-
-        root_backend = get_qml_root_backend()
-        backend = root_backend.findChild(QtCore.QObject, 'qml_table_view_backend_{}'.format(context_id))
-        assert backend
-        item_view = ItemViewProxy(backend)
-
-        new_model.setParent(item_view)
-
-        list_gui_context.item_view = item_view
-        list_gui_context.view = backend.property('view')
+        response = qml_action_step(list_gui_context, action_step_name,
+                serialized_step, { 'model': new_model }, model=new_model)
 
+        # FIXME: is this needed?
+        # - actions states have already been serialized
+        # - set_filter changes action states?
         UpdateActions().gui_run(list_gui_context)
```

Comment:
mogelijk is dit inderdaad nt meer nodig
(vroeger waren die action/filter states zetten een kluwen vanwege de volgorde van request/response waarmee de hele view werd opgebouwd.  probleem is dat de states vd filter de value (query) bepalen, maar die hele problemathiek zou nu eenvoudiger moeten kunnen omdat de boel eerst w bepaald, en dan geserialiseerd.)

Reply:
Heb getest met enkele views waarbij de intitieele filters het aantal rijen beperkt (t.o.v. alle rijen) en dit lijkt ook te werken zonder deze call.